### PR TITLE
Addresses Issue #1866 for render_to() to return a position rect

### PR DESCRIPTION
**Pull request by mmurrs**
The FX6_TRUNC was resulting in incorrect values caused by bit-shifting

### CODE REVIEW

**Comment by MyreMylar** [comment]
Amazing!

The main thing we need here to finish this off is a unit test of the functionality here:

https://github.com/pygame/pygame/blob/48571a6b060b5c528ba584684e9120cec96afa99/test/freetype_test.py#L661-L749

Currently there are some commented out lines in there that refer to the returned rectangle but obviously (since the test passes) there are no actual checks of the supposed functionality. Ideally we could make an understandable change that would make this test fail on the current master but then work with your new fix.

**Comment by mmurrs** [comment]
Ok thanks for the feedback, I will go through those unit tests to make sure render_to() is working properly.

File: test/freetype_test.py [hunk 1]
```diff
@@ -669,7 +669,9 @@ def test_freetype_Font_render_to(self):
         rrect = font.render_to(surf, (32, 32), "FoobarBaz", color, None, size=24)
         self.assertIsInstance(rrect, pygame.Rect)
         self.assertEqual(rrect.top, rrect.height)
-        ## self.assertEqual(rrect.left, something or other)
+        ##test that render_to is position
+        self.assertEqual(rrect.left, 32)
+        self.assertEqual(rrect.right, 32)
```
Comment (MyreMylar) [inline test/freetype_test.py:674]: This seems wrong, the returned rect (according to the docs) is supposed to have the size as well as the position of the rendered text - so the right hand side of the rectangle should definitely be further to the right right than the left hand side with the text 'FoobarBaz' at font size 24.

**Example picture:**

image

It won't come out to exactly 156px on the right hand side of the rect as that's m,e just eyeballing it in a paint program, but it should definitely have at least _some_ width.

File: test/freetype_test.py [hunk 2]
```diff
@@ -680,7 +682,8 @@ def test_freetype_Font_render_to(self):
         self.assertEqual(rrect.top, rrect.height)
         self.assertNotEqual(rrect.size, rect.size)
         rrect = font.render_to(surf, (20.1, 18.9), "FoobarBax", color, None, size=24)
-        ## self.assertEqual(tuple(rend[1].topleft), (20, 18))
+        # Added the test back
+        self.assertEqual(tuple(rend[1].topleft), (20, 18))
```
Comment (MyreMylar) [inline test/freetype_test.py:686]: Could also check the bottomright here once the width & height part is working.

**Comment by MyreMylar** [comment]
Looks like no progress on this one. Gonna just close it up, no hard feelings, just useful to keep a clean-ish list of open PRs. You can always re-open it later if you have new inspiration.

**Pull request closed without merging**